### PR TITLE
Drop py27 and py33 support

### PR DESCRIPTION
**Pull request by mshriver**
Remove py27 and py33 from tox, travis, packaging
Clean up imports that were different between python versions
remove requirements.txt, move to setup.py
remove uses of six library for direct py3 built-ins

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
-envlist = py{27,33,34,35,36,37},codechecks
+envlist = py{35,36,37},codechecks
 
 [testenv]
 deps=
     pytest
     pytest-cov
     coveralls
     mock
+    parsedatetime
```
Comment (mshriver) [inline tox.ini]: I don't like doubly defining this, trying to figure out why tox isn't pulling it from setup.py - even when `python setup.py develop` is explicitly called in tox.

**Comment by izapolsk** [comment]
@quarckster, cool, I was also planning to add asyncio patch to wait_for. what's your opinion about that tenacity library ?

**Comment by quarckster** [comment]
@izapolsk It can do the same as `wait_for` and even more but the usage is quite different.

**Comment by mshriver** [comment]
@quarckster thanks for the heads up!

**Pull request merged**